### ISSUE TITLE
Play encrypted videos using AES

### ISSUE THREAD

**Issue by mhdkadi** [media-kit/media-kit#565]
We are developing an Android/IOS app that plays online videos that are stored on DigitalOcean Spaces
The videos are encrypted using AES so is there a way to play those videos using media_kit player and decrypt the video in real  time?

**Comment by 23doors** [comment]
Dunno if anyone actually checked if it works, but you can try this: https://github.com/media-kit/media-kit/issues/381#issuecomment-1685260863

**Comment by mhdkadi** [comment]
is the player natively can decrypt an encoded AES video ?

**Comment by 23doors** [comment]
No idea what you mean by that. Mobile native players and web players when used with DASH/HLS/ISM supports AES CBCS and CTR encryptions where key is acquired by either standarized licensing methods (Widevine, Playready, Fairplay, Clear key) or a custom method. A lot here depends on platform used as normally Android lean more towards DASH/HLS+Widevine. And Apple ecosystem towards HLS+Fairplay. 

If you know a decryption key or have custom way of delivering it, you can use API I linked. If you want to use full licensing process from device, there is no player that will fit all platforms and all needs, especially on desktop platforms where you have no system API for licensing. 

Basically, do your own research of what's needed first.

**Comment by mhdkadi** [comment]
### My Code Is

```
class VideoPlayerWidgetTest extends StatefulWidget {
  const VideoPlayerWidgetTest({super.key});
  @override
  State<VideoPlayerWidgetTest> createState() => _VideoPlayerWidgetState();
}

class _VideoPlayerWidgetState extends State<VideoPlayerWidgetTest> {
  final player = Player(configuration: const PlayerConfiguration());
  late final VideoController controller = VideoController(player);

  StreamSubscription<String>? errorStream;
  @override
  void initState() {
    super.initState();

    errorStream = player.stream.error.listen((error) async {
      logger(error, name: "Video Player");
    });

    playVideo();
  }

  @override
  void dispose() {
    player.dispose();
    errorStream?.cancel();
    super.dispose();
  }

  Future<void> playVideo() async {
    if (player.platform is NativePlayer) {
      await (player.platform as dynamic).setProperty(
        'demuxer-lavf-o',
        'decryption_key=0123456789abcdef0123456789abcdef',
      );
    }
    await player.open(Media(
        'https://i-knwoeldge.fra1.cdn.digitaloceanspaces.com/big_buck_bunny_480p_trailer.m4v'));
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(body: Video(controller: controller));
  }
}
```

### Error Log
`[Video Player] Failed to recognize file format.`

**Comment by mhdkadi** [comment]
Is there any answer?

**Comment by mhdkadi** [comment]
any help?

**Comment by alexmercerind** [comment]
> Any help?

If you can tell any other application or library where this exact setup (key & URL) works.

I don't have any specialization in this either.
